smarty/forcefield.py

```diff
@@ -389,9 +386,144 @@ def registerGenerator(self, generator):
         """Register a new generator."""
         self._forces.append(generator)
 
+    def getParameter(self, smirks = None, paramID=None, force_type='Implied'):
+        """Get info associated with a particular parameter as specified by SMIRKS or parameter ID, and optionally force term.
+
+    Parameters
+    ----------
+    smirks (optional) : str
+        Default None. If specified, will pull parameters on line containing this `smirks`.
+    paramID : str
+        Default None. If specified, will pull parameters on line with this `id` 
+    force_type : str
+        Default "Implied". Optionally, specify a particular force type such as 
+        "HarmonicBondForce" or "HarmonicAngleForce" etc. to search for a 
+        matching ID or SMIRKS. 
+    
+
+    Returns
+    -------
+    params : dict
+        Dictionary of attributes (parameters and their descriptions) from XML    
+
+
+Usage notes: SMIRKS or parameter ID must be specified.
+
+To do: Update behavior of "Implied" force_type so it raises an exception if the parameter is not uniquely identified by the provided info.
+"""
+        # Check for valid input
+        if smirks and paramID:
+            raise ValueError("Error: Specify SMIRKS OR parameter ID but not both.")
+        if smirks==None and paramID==None:
+            raise ValueError("Error: Must specify SMIRKS and/or parameter ID.")
```

Comment:
You raise an error if both smirks and paramID are specified, doesn't that mean the Error message here should say "SMIRKS or parameter..." not "and/or"?